includes/Modules/Analytics_4.php

```diff
@@ -1331,6 +1340,58 @@ protected function create_analytics_site_data_request( $property_id, array $args
 		return $request;
 	}
 
+	/**
+	 * Validates the given metrics for a report.
+	 *
+	 * Metrics must have valid names, matching the regular expression ^[a-zA-Z0-9_]+$ in keeping with the GA4 API.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Google_Service_AnalyticsData_Metric[] $metrics The metrics to validate.
+	 * @throws Invalid_Report_Metrics_Exception Thrown if the metrics are invalid.
+	 */
+	protected function validate_report_metrics( $metrics ) {
```

Comment:
The `validate_report_metrics()` method mentioned in the IB has been renamed to `validate_shared_report_metrics()` and is only called for a shared data request, so we don't want to add the metric name validation to that method as it needs to be called for all requests. Therefore I've added `validate_report_metrics()` again to cover the validation for all requests.